tests/test_subpixmaxima.py

```diff
@@ -49,13 +50,117 @@ def test_subpixmaxima():
+    dataset = HeatmapDataset(
+        root_directory="toy_datasets/toymouseRunningData",
+        csv_path="CollectedData_.csv",
+        header_rows=[1, 2],
+        imgaug_transform=imgaug_transform,
+    )
+    test_img, gt_heatmap, gt_keypts = dataset.__getitem__(idx=0)
+    assert(gt_keypts.shape == (torch.Size([34])))
+    gt_heatmap = gt_heatmap.unsqueeze(0)
+    gt_keypts = gt_keypts.unsqueeze(0).reshape(1, 17, 2)
+    assert(gt_heatmap.shape == (1, 17, 96, 96))
+    torch_heatmap = generate_heatmaps(gt_keypts, height=384, width=384, output_shape=(96,96))
+    SubPixMax = SubPixelMaxima(
+        output_shape=(96, 96),
+        output_sigma=torch.tensor(1.25, device=_TORCH_DEVICE),
+        upsample_factor=torch.tensor(2, device=_TORCH_DEVICE),
+        coordinate_scale=torch.tensor(4, device=_TORCH_DEVICE),  # 2 ** 2
+        confidence_scale=torch.tensor(1, device=_TORCH_DEVICE), 
+        threshold=None,
+        device=_TORCH_DEVICE
+    )
+    og_maxima, gt_confidence = SubPixMax.run(gt_heatmap.to(_TORCH_DEVICE))
+    torch_maxima, confidence_t = SubPixMax.run(torch_heatmap.to(_TORCH_DEVICE))
+    assert(og_maxima == torch_maxima).all()
+    assert(gt_confidence == confidence_t).all()
+
+def test_generate_keypoints_weird_shape():
+    data_transform = []
+    OG_SHAPE = (384, 256)
+    DOWNSAMPLE_FACTOR = 2
+    output_shape = (OG_SHAPE[0]//(2** DOWNSAMPLE_FACTOR), OG_SHAPE[1]//(2** DOWNSAMPLE_FACTOR))
+    data_transform.append(
+        iaa.Resize({"height": OG_SHAPE[0], "width": OG_SHAPE[1]})
+    )  # dlc dimensions need to be repeatably divisable by 2
+    imgaug_transform = iaa.Sequential(data_transform)
+    dataset = HeatmapDataset(
+        root_directory="toy_datasets/toymouseRunningData",
+        csv_path="CollectedData_.csv",
+        header_rows=[1, 2],
+        imgaug_transform=imgaug_transform,
+    )
+    test_img, gt_heatmap, gt_keypts = dataset.__getitem__(idx=0)
+    assert(gt_keypts.shape == (torch.Size([34])))
+    gt_heatmap = gt_heatmap.unsqueeze(0)
+    gt_keypts = gt_keypts.unsqueeze(0).reshape(1, 17, 2)
+    assert(gt_heatmap.shape == (1, 17, output_shape[0], output_shape[1]))
+    torch_heatmap = generate_heatmaps(gt_keypts, height=OG_SHAPE[0], width=OG_SHAPE[1], output_shape=output_shape)
+    SubPixMax = SubPixelMaxima(
+        output_shape=output_shape,
+        output_sigma=torch.tensor(1.25, device=_TORCH_DEVICE),
+        upsample_factor=torch.tensor(2, device=_TORCH_DEVICE),
+        coordinate_scale=torch.tensor(4, device=_TORCH_DEVICE),  # 2 ** 2
+        confidence_scale=torch.tensor(1, device=_TORCH_DEVICE), 
+        threshold=None,
+        device=_TORCH_DEVICE
+    )
+    og_maxima, gt_confidence = SubPixMax.run(gt_heatmap.to(_TORCH_DEVICE))
+    torch_maxima, confidence_t = SubPixMax.run(torch_heatmap.to(_TORCH_DEVICE))
+    assert(og_maxima == torch_maxima).all()
+    assert(gt_confidence == confidence_t).all()
+
+def test_generate_keypoints_batched():
```

Comment:
maybe provide a description about this test? suggestion:
"comparing the ground truth heatmaps generated by numpy with the heatmaps generated from torch. TODO: convert everything to torch".